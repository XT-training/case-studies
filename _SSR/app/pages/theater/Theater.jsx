import React, { Component } from 'react';
import { connect } from 'react-redux';
import Page from '../Page';
import commonUtil from '../../utils/commonUtil';
import TheaterContainer from '../../containers/theater/TheaterContainer';
import { IntlProvider } from 'react-intl';

class Theatre extends Component {
    constructor(props) {
        super(props);
    }
    getMetaData() {
        return {
            title: this.pageTitle(),
            meta: this.pageMeta(),
            link: this.pageLink(),
            labels: this.props.labels
        };
    }
    // TODO:: title,meta,link function should be part of parent class.
    pageTitle = () => 'Title';

    pageMeta = () => {
        [];
    };

    pageLink = () => {
        [];
    };

    render() {
        return (
            <Page {...this.getMetaData()}>
                <TheaterContainer {...this.props} />
            </Page>
        );
    }
}

const mapStateToProps = state => ({
    theaters: state.theaters,
    labels: state.app.labels
});

export default connect(mapStateToProps)(Theatre);
