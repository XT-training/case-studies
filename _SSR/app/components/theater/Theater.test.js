import { mount, shallow, render } from 'enzyme';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Theater from './Theater';

configure({ adapter: new Adapter() });

describe('Theater', () => {
    it('Theater should render correctly', () => {
        const theater = {
            theatreName: 'phoneix',
            theatreAddress: 'abc',
            _id: 10
        };

        const component = shallow(<Theater theater={theater} />);

        expect(component).toMatchSnapshot();
    });
});
