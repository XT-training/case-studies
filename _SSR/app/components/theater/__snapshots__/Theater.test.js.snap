// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theater Theater should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Theater
    theater={
      Object {
        "_id": 10,
        "theatreAddress": "abc",
        "theatreName": "phoneix",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="name"
        >
          <Link
            to="/screening/10"
          >
            phoneix
          </Link>
        </div>,
        <div
          className="address"
        >
          abc
        </div>,
      ],
      "className": "theater",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            to="/screening/10"
          >
            phoneix
          </Link>,
          "className": "name",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "phoneix",
            "to": "/screening/10",
          },
          "ref": null,
          "rendered": "phoneix",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "abc",
          "className": "address",
        },
        "ref": null,
        "rendered": "abc",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="name"
          >
            <Link
              to="/screening/10"
            >
              phoneix
            </Link>
          </div>,
          <div
            className="address"
          >
            abc
          </div>,
        ],
        "className": "theater",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              to="/screening/10"
            >
              phoneix
            </Link>,
            "className": "name",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "phoneix",
              "to": "/screening/10",
            },
            "ref": null,
            "rendered": "phoneix",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "abc",
            "className": "address",
          },
          "ref": null,
          "rendered": "abc",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
